loama/src/lib/state.ts

```diff
@@ -1,13 +1,58 @@
 import { ref } from "vue";
 import type { Entry } from "./types";
-import { activeController } from "loama-controller";
+import { activeController, type PublicSubject, type ResourcePermissions, type WebIdSubject } from "loama-controller";
+import { defineStore } from "pinia";
+import { store } from "loama-app";
+import { uriToName } from "@/lib/utils"
+import { PhLock, PhLockOpen } from "@phosphor-icons/vue";
 
-export const selectedEntry = ref<Entry | null>(null)
 
-export const refreshEntryPermissions = async () => {
-    if (!selectedEntry.value) {
-        throw new Error('No selected entry to update permissions for');
-    }
-    const newResourceInfo = await activeController.getResourcePermissionList(selectedEntry.value.resourceUrl);
-    selectedEntry.value.permissionsPerSubject = newResourceInfo.permissionsPerSubject;
+type PodStore = {
+    podEntries: ResourcePermissions<WebIdSubject | PublicSubject>[];
+    selectedEntry: Entry | null;
 }
+
+export const usePodStore = defineStore("pod", {
+    state: (): PodStore => ({
+        selectedEntry: null,
+        podEntries: [],
+    }),
+    getters: {
+        formattedEntries(state) {
+            return state.podEntries.map(resourcePermissions => {
+                const uri = resourcePermissions.resourceUrl.replace(store.usedPod, '');
+                const isContainer = uri.charAt(uri.length - 1) === '/'
+                const name = uriToName(uri, isContainer);
+                const publicAccess = (resourcePermissions.permissionsPerSubject.find(s => s.subject.type === "public")?.permissions.length ?? 0) > 0;
+                return {
+                    isContainer,
+                    name,
+                    public: {
+                        access: publicAccess,
+                        icon: (publicAccess) ? PhLockOpen : PhLock
+                    },
+                    ...resourcePermissions
+                }
+            })
+        }
+    },
+    actions: {
+        async loadThings(url: string) {
```

Comment:
please don't use 'thing', I think 'resource' is used throughout the codebase?

Reply:
Ah, I reused the name of the function in the component, i've renamed it